bpf/process/regs.h

```diff
@@ -133,4 +133,17 @@ write_reg(struct pt_regs *ctx, __u32 dst, __u8 size, __u64 val)
 }
 
 #endif /* __TARGET_ARCH_x86 */
+
+#if defined(__TARGET_ARCH_arm64)
+FUNC_INLINE __u64
+read_reg(struct pt_regs *ctx, __u32 src, __u8 shift)
+{
+	unsigned long val = (unsigned long)-1UL, off = src & 0xfff;
+
+	probe_read_kernel(&val, sizeof(val), (void *)ctx + off);
```

Comment:
Q: previous code was error checking the `probe_read_kernel` call; do we want to restore it?

Reply:
hm I thought it's not used.. we do not test 'set' action on arm right? val will be 0 on error, which is likely to fail on write_user_arg, but you're right we should check it

I wonder if we could add read_reg/write_reg interface for arm.. and 'set' action test